WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -414,35 +414,41 @@ private extension DashboardViewModel {
                               canShowAnalytics: Bool) async {
         dashboardCards = await {
             if let stored = await loadDashboardCards() {
-                return stored
+                var mutableStored = stored
+                // Remove saved analytics cards if they should not be shown anymore.
+                if !canShowAnalytics {
+                    mutableStored.removeAll { $0.type == .performance || $0.type == .topPerformers }
+                }
+                return mutableStored
             } else {
+                // Start with default values, cards will be removed further below as needed.
                 return [DashboardCard(type: .onboarding, enabled: canShowOnboarding),
                         DashboardCard(type: .performance, enabled: canShowAnalytics),
                         DashboardCard(type: .topPerformers, enabled: canShowAnalytics),
                         DashboardCard(type: .blaze, enabled: canShowBlaze)]
             }
         }()
 
-        unavailableDashboardCards = []
-
-        if let performanceCard = dashboardCards.first(where: { $0.type == .performance }),
-            !canShowAnalytics {
-            unavailableDashboardCards.append(performanceCard)
+        // If should not be shown, ensure Onboarding is not visible on Dashboard
+        if !canShowOnboarding {
+            dashboardCards.removeAll { $0.type == .onboarding }
         }
 
-        if let topPerformersCard = dashboardCards.first(where: { $0.type == .topPerformers }),
-            !canShowAnalytics {
-            unavailableDashboardCards.append(topPerformersCard)
+        // If should not be shown, ensure Blaze is not visible on Dashboard
+        if !canShowBlaze {
+            dashboardCards.removeAll { $0.type == .blaze }
         }
 
-        if let onboardingCard = dashboardCards.first(where: { $0.type == .onboarding }),
-           !canShowOnboarding {
-            unavailableDashboardCards.append(onboardingCard)
-        }
+        // Manually update the saved cards because it might have been changed above.
+        didCustomizeDashboardCards(dashboardCards)
```

Comment:
@hafizrahman Are you sure this is necessary? Also I added a tracking event in `didCustomizeDashboardCards` so calling it will track a manual save - you might want to call a separate method instead.

Reply:
That's incorrectly added, I've removed it in https://github.com/woocommerce/woocommerce-ios/pull/12572/commits/188747afb14dee4605417efea74b4098044d4924